flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -173,39 +157,49 @@ public static void startCassandra() throws IOException {
 			cassandra.start();
 		}
 
-		try {
-			Thread.sleep(1000 * 10);
-		} catch (InterruptedException e) { //give cassandra a few seconds to start up
+		// give cassandra a few seconds to start up
+		long start = System.currentTimeMillis();
+		long deadline = start + 1000 * 10;
+		while (System.currentTimeMillis() < deadline) {
+			try {
+				Thread.sleep(1000);
+			} catch (InterruptedException ignored) {
+			}
 		}
 
-		cluster = builder.getCluster();
-		session = cluster.connect();
+		// start establishing a connection within 30 seconds
+		start = System.currentTimeMillis();
+		deadline = start + 1000 * 30;
+		while (true) {
+			try {
+				cluster = builder.getCluster();
+				session = cluster.connect();
+				break;
+			} catch (Exception e) {
+				if (System.currentTimeMillis() > deadline) {
+					throw e;
+				}
+				try {
+					Thread.sleep(2000);
```

Comment:
How about reducing this to, say 500ms, to make it react a bit sooner once the connection can be established.